rfcs/2022-04-20-12187-log-namespacing.md

```diff
@@ -0,0 +1,399 @@
+
+```json
+{
+  "source_type": "datadog_agent",
+  "ingest_timestamp": "2022-04-14T19:14:21.899623781Z"
+}
+```
+
+-----
+
+#### Datadog Agent source / bytes codec / Vector namespace
+
+event
+
+```json
+{
+  "data": "{\"proportional\":702036423,\"integral\":15089925750456892008,\"derivative\":-6.4676193438086e263,\"vegetable\":20003,\"mineral\":\"vsd5fwYBv\"}",
+  "ddsource": "waters",
+  "ddtags": "env:prod",
+  "hostname": "beta",
+  "service": "cernan",
+  "status": "notice",
+  "timestamp": "2066-08-09T04:24:42.1234Z"
+}
+```
+
+metadata
+
+```json
+{
+  "source_type": "datadog_agent",
+  "ingest_timestamp": "2022-04-14T19:14:21.899623781Z"
+}
+```
+
+-----
+
+#### Kafka source / json codec / Vector namespace
+
+event
+
+```json
+{
+  "data": {
+    "derivative": -2.266778047142367e+125,
+    "integral": "13028769352377685187",
+    "mineral": "H 9 ",
+    "proportional": 3673342615,
+    "vegetable": -30083
+  },
+  "key": "the key of the message"
+  // headers were originally nested under a configurable "headers_key". This is using a static value.
+  "headers": {
+    "header-a-key": "header-a-value",
+    "header-b-key": "header-b-value",
+  }
+}
+```
+
+metadata
+
+```json
+{
+  "topic": "name of topic",
+  "partition": 3,
+  "offset": 1829448,
+  "source_type": "kafka",
+  "ingest_timestamp": "2022-04-14T19:14:21.899623781Z"
+}
+```
+
+-----
+
+#### Kubernetes Logs / Vector namespace
+
+event (a string as the root event element)
+
+```text
+F1015 11:01:46.499073       1 main.go:39] error getting server version: Get \"https://10.96.0.1:443/version?timeout=32s\": dial tcp 10.96.0.1:443: connect: network is unreachable
+```
+
+metadata
+
+```json
+{
+  "file": "/var/log/pods/kube-system_storage-provisioner_93bde4d0-9731-4785-a80e-cd27ba8ad7c2/storage-provisioner/1.log",
+  "container_image": "gcr.io/k8s-minikube/storage-provisioner:v3",
+  "container_name": "storage-provisioner",
+  "namespace_labels": {
+    "kubernetes.io/metadata.name": "kube-system"
+  },
+  "pod_annotations": {
+    "prometheus.io/scrape": "false"
+  },
+  "pod_ip": "192.168.1.1",
+  "pod_ips": [
+    "192.168.1.1",
+    "::1"
+  ],
+  "pod_labels": {
+    "addonmanager.kubernetes.io/mode": "Reconcile",
+    "gcp-auth-skip-secret": "true",
+    "integration-test": "storage-provisioner"
+  },
+  "pod_name": "storage-provisioner",
+  "pod_namespace": "kube-system",
+  "pod_node_name": "minikube",
+  "pod_uid": "93bde4d0-9731-4785-a80e-cd27ba8ad7c2",
+  "stream": "stderr",
+  "source_type": "kubernetes_logs",
+  "ingest_timestamp": "2020-10-15T11:01:46.499555308Z"
+}
+```
+
+-----
+
+
+#### Syslog / Vector namespace
+
+event (the "data" key was elided)
+
+```json
+{
+    "message": "Hello Vector",
+    "hostname": "localhost",
+    "severity": "info",
+    "facility": "facility",
+    "appname": "Vector Hello World",
+    "msgid": "238467-435-235-a3478fh",
+    "procid": 13512,
+    // this name is up for debate. Arbitrary keys need to be nested under something though
```

Comment:
That might work. I was mainly suggesting that the smaller details like this can be decided in the individual PRs for each source, rather than in the RFC.